lib/jormungandr/test/integration/Main.hs

```diff
@@ -122,19 +124,21 @@ specWithServer
     :: (CM.Configuration, Trace IO Text)
     -> SpecWith (Context (Jormungandr 'Testnet))
     -> Spec
-specWithServer logCfg = beforeAll start . after tearDown
+specWithServer logCfg = beforeAll start . afterAll _cleanup . after tearDown
   where
     start :: IO (Context (Jormungandr 'Testnet))
     start = do
         ctx <- newEmptyMVar
-        pid <- async $ withConfig $ \jmCfg -> do
+        finished <- newEmptyMVar
+        let untilFinished = fmap (either id id) . race (takeMVar finished)
+        pid <- async $ withConfig $ \jmCfg -> untilFinished $ do
```

Comment:
suggestion:

```hs
         let untilFinished = race (takeMVar finished)
         pid <- async $ withConfig $ \jmCfg -> void $ untilFinished $ do
```

?

Reply:
I see what you're getting at though. It also bothered me ⇒ #885.